libs/state-machine/src/lib/step-find-functions.ts

```diff
@@ -0,0 +1,86 @@
+import { StateMachineError, StateMachineErrorCode } from './StateMachineError';
+import type { StepFindFunction, WizardLayout, WizardStepLayout } from './types';
+
+const isNormalStep = (step: WizardStepLayout) =>
+	step.parentStepId === undefined && step.role === undefined;
+
+const isNormalStepOrEditStart = (step: WizardStepLayout) =>
+	step.parentStepId === undefined &&
+	(step.role === undefined || step.role === 'EDIT_START');
+
+const isNormalStepOrCreateStart = (step: WizardStepLayout) =>
+	step.parentStepId === undefined &&
+	(step.role === undefined || step.role === 'CREATE_START');
+
+const getContext = (wizard: WizardLayout, step: WizardStepLayout) => {
+	const allIds: string[] = Object.keys(wizard);
+	const allSteps: WizardStepLayout[] = Object.values(wizard);
+	const indexOfCurrentStep = allSteps.indexOf(step);
+	if (indexOfCurrentStep === -1) {
+		throw new StateMachineError(
+			'step is not part of wizard',
+			StateMachineErrorCode.NoSuchStep,
+			true,
+		);
+	}
+	return { allIds, allSteps, indexOfCurrentStep };
+};
+
+export const goToNextNormalStep: StepFindFunction = (wizard, step) => {
```

Comment:
these functions are returning the id of the next logical step - wondering if the name might reflect that? something like `getNextStepId` or similar

Reply:
👍 that sounds good - it's a better description of function itself.